### PR TITLE
Install IBC-Hooks to be able to invoke CosmWasm contracts from the other chains

### PR DESCRIPTION
**Pull request by artur-abliazimov**
## Summary by CodeRabbit

- **New Features**
  - Enabled calling CosmWasm contracts from other chains using IBC-Hooks.
  
- **Updates**
  - Updated Golang version to 1.22.
  - Upgraded multiple dependencies to their latest versions for improved performance and security.

- **Improvements**
  - Enhanced module integration for seamless interaction with IBC-related functionalities.
  - Added new IBC middleware components to the application for better compatibility and functionality.

### CODE REVIEW

File: warden/app/app.go [hunk 1]
```diff
@@ -82,6 +83,14 @@ const (
 var (
 	// DefaultNodeHome default home directories for the application daemon
 	DefaultNodeHome string
+
+	// ModuleBasics defines the module BasicManager is in charge of setting up basic,
+	// non-dependant module elements, such as codec registration
+	// and genesis verification.
+	ModuleBasics = module.NewBasicManager(
+		ibchooks.AppModuleBasic{},
```
Comment (pitasi) [inline warden/app/app.go]: I think this should be added here instead:
https://github.com/warden-protocol/wardenprotocol/blob/f66d93d8518230b2719a9c0c0f46b118e4420f87/warden/app/app.go#L177

can you check if it works?

It should also make unnecessary to call `ModuleBasics.RegisterGRPCGatewayRoutes`

Comment (artur-abliazimov) [reply warden/app/app.go]: Fixed. Thanks!

**Review by pitasi — DISMISSED** [review]
utACK 😁

**Comment by artur-abliazimov** [comment]
> utACK 😁

Yeah, me neither =) I think we will test it after deployment to Buenavista.

**Review by pitasi — CHANGES_REQUESTED** [review]
forgot to ask you: could you squash all commits into one? I think it's sufficient

**Review by pitasi — APPROVED** [review]

**Pull request merged**